addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -226,36 +221,54 @@ function factory(dependencies) {
                 return;
             }
 
-            // If the channel wasn't known, joining the channel already updated
-            // the counter.
-            if (!channel) {
-                return;
-            }
-
             // Message from mailing channel should not make a notification in
             // Odoo for users with notification "Handled by Email".
             // Channel has been marked as read server-side in this case, so
             // it should not display a notification by incrementing the
             // unread counter.
             if (
-                channel &&
-                channel.mass_mailing &&
+                message.originThread.mass_mailing &&
```

Comment:
`message.originThread` could be `undefined`, at least design-wise it's not guaranteed to be set (and it's clean to assume that).

Reply:
`originThread` is guaranteed to be set here, it's the `_handleNotificationChannelMessage` for which channel is definitely known and explicit inserted into message data at the top of the method.

Reply:
Which might actually be a problem now that I put this on paper.

Reply:
Yes very annoying problem actually. No channel should ever be inserted if it does not have the full data from channel_info.
Important information such as `uuid` must be "required", or it can lead to issues further down (when folding chat window), missing name leads to badly named chat window, etc.

We'll need to discuss how we can handle that.